validation/validate_document.py

```diff
@@ -0,0 +1,57 @@
+import filetype
+from django.core.exceptions import ValidationError
+
+
+def validate_document_file(file):
+    """
+    Validates an uploaded document file for:
+    - Allowed file extensions
+    - Allowed MIME types
+    - Maximum file size (20MB)
+
+    Args:
+        file (File): The uploaded file to validate.
+
+    Raises:
+        ValidationError: If the file is invalid based on extension, MIME type, or size.
+    """
+
+    allowed_extensions = [
+        "pdf", "doc", "docx", "txt", "odt", "rtf",
+        "xls", "xlsx", "ppt", "pptx", "zip", "rar"
+    ]
+    ext = file.name.split('.')[-1].lower()
```

Comment:
Split filename on `'.'` from the right so names like `file.tar.gz` are handled more robustly:
```
ext = file.name.rsplit('.', 1)[-1].lower()
```